3-0-java-core/3-6-4-random-field-comparator/src/main/java/com/bobocode/se/RandomFieldComparator.java

```diff
@@ -0,0 +1,82 @@
+package com.bobocode.se;
+
+import com.bobocode.util.ExerciseNotCompletedException;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Objects;
+import lombok.SneakyThrows;
+
+/**
+ * A generic comparator that is comparing a random field of the given class. The field is either primitive or
+ * {@link Comparable}. It is chosen during comparator instance creation and is used for all comparisons.
+ * <p>
+ * If no field is available to compare, the constructor throws {@link IllegalArgumentException}
+ *
+ * @param <T> the type of the objects that may be compared by this comparator
+ *<p><p>
+ *  <strong>TODO: to get the most out of your learning, <a href="https://www.bobocode.com/learn">visit our website</a></strong>
+ *  <p>
+ *
+ * @author Stanislav Zabramnyi
+ */
+public class RandomFieldComparator<T> implements Comparator<T> {
+
+    private final Class<T> targetType;
+    private final Field fieldToCompare;
+
+    public RandomFieldComparator(Class<T> targetType) {
+        Objects.requireNonNull(targetType);
+        this.targetType = targetType;
```

Comment:
It makes sense to do  `this.targetType = Objects.requireNonNull(targetType);` or even
`this.targetType = requireNonNull(targetType);`